src/lib/sw.js

```diff
@@ -149,6 +117,62 @@ workboxRouting.registerRoute(
   new workboxStrategies.StaleWhileRevalidate(),
 );
 
+/**
+ * Match "/foo-bar/ "and "/foo-bar/as/many/of-these-as-you-like/" (with optional trailing
+ * "index.html"), normal page nodes for web.dev. This only matches on pathname.
+ *
+ * This fetch handler internally fetches the required partial using `partialStrategy`, and
+ * generates the page's real HTML based on the layout template.
+ */
+const contentPathRe = new RegExp("^(/(?:[\\w-]+/)*)(?:|index\\.html)$");
+self.addEventListener("fetch", (event) => {
```

Comment:
Curious as to whether there's a reason you went with a "vanilla" `fetch` handler here instead of using `workboxRouting.registerRoute()`. That's not a problem in general, but it could get confusing at some point in the future, because 

1. `workboxRouting.setCatchHandler()` won't be invoked for failures here (maybe that's intentional?)
2. If there are any calls to `workboxRouting.registerRoute()` that happen in this service worker after this `fetch` handler declaration, they'll end up taking precedence over your `fetch` handler, since they'll be triggered as part of `workbox-routing`'s (earlier) `fetch` handler.

I'm fairly sure this could be rewritten to use `workboxRouting.registerRoute()` with a custom [`matchCallback`](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-routing#~matchCallback) and [`handlerCallback`](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-routing#~handlerCallback) if you were open to that.

If your goal is not to trigger the `workboxRouting.setCatchHandler()` (point 1), then wrapping an `async` `handlerCallback` in `try`/`catch` and returning a properly formatted `Response.error()` in the `catch` block should be sufficient to avoid that.

Reply:
So 1. isn't important as even in the offline case, this fetch handler should return safely. If there's any other error we are happy to actually give up.

2. I've rewritten this handler (and the other remaining) to use `registerRoute`. It's a much nicer interface than the odd `repsondWith` dance.